src/main/java/seedu/address/logic/commands/AddContactCommand.java

```diff
@@ -14,10 +14,12 @@
 /**
  * Adds a person to the address book.
  */
-public class AddCommand extends Command {
+public class AddContactCommand extends Command {
 
     public static final String COMMAND_WORD = "add";
 
+    public static final String COMMAND_TYPE = "/c";
```

Comment:
Does each class really need to know the `COMMAND_TYPE` string? It seems like it would be better to just have the AddressBookParser store the main `COMMAND_TYPE` string variants. Either that, or these classes should be taking the AddressBookParser's `COMMAND_TYPE` definition. Because right now, to change the `COMMAND_TYPE` string, every single file has to be edited

Reply:
And it seems that the `COMMAND_TYPE` String is largely not used except to reconstruct the equivalent user input

Reply:
hmm, interestingly this is also something that i had issues with. i avoided using `AddressBookParser.COMMAND_TYPE` as i did not want to create further cyclic dependencies (`AddContactCommandParser` -> `AddContactCommand` -> `AddressBookParser` -> `AddContactCommandParser`).

a plausible solution would to be to create an enumeration to only store command words/types and for all classes to reference that instead.

Reply:
My suggestion was actually to not have this at all. But the enum-type would probably be the second best alternative as both `AddressBookParser` and the relevant commands will depend on the String specified in the enum-type. Because as is right now, the current implementation is the equivalent of the cyclic dependency you mentioned, just not strictly coded in. If you change `AddressBookParser`'s string to detect `Contact`-type commands, then you'll still have to change those within each `Contact`-type command class.

Reply:
the implementation would be to create two enumerations - `CommandWord` and `CommandType` - and have `AddressBookParser` and the Command classes refer to the enumeration instead. this accomplishes two things:

1. resolves the aforementioned cyclic dependencies.
1. test cases can remain as is, since they currently refer to each individual Command class' word/type.

i think there is merit in having the test cases remain in their current classes as they are intended to test certain classes.

> My suggestion was actually to not have this at all.

hmm, what would you propose then?

Reply:
I think only one enumeration is sufficient (CommandType). Because 'add' is a unique identifier for the AddContactCommand, or at least it was. But I feel like the reason both `AddressBookParser` and the individual command classes should not hold the contact identifier's specific implementation within them is that it is a property that they all share and depend on. So a separate enumeration would fit this. 
 
I change my mind regarding holding references within the command classes themselves to their CommandType. It does make recreating the command for testing easier.

> i think there is merit in having the test cases remain in their current classes as they are intended to test certain classes.
I don't think this is really relevant to the discussion because the individual command classes don't actually use their own command word and command type for testing. Only `AddressBookParserTest` uses them to recreate it.